src/ib.cc

```diff
@@ -50,9 +50,21 @@ IbMr::IbMr(ibv_pd* pd, void* buff, std::size_t size) : buff(buff) {
   }
   uintptr_t addr = reinterpret_cast<uintptr_t>(buff) & -pageSize;
   std::size_t pages = (size + (reinterpret_cast<uintptr_t>(buff) - addr) + pageSize - 1) / pageSize;
-  this->mr = IBVerbs::ibv_reg_mr2(pd, reinterpret_cast<void*>(addr), pages * pageSize,
-                                  IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_WRITE | IBV_ACCESS_REMOTE_READ |
-                                      IBV_ACCESS_RELAXED_ORDERING | IBV_ACCESS_REMOTE_ATOMIC);
+
+  CUdeviceptr dptr = reinterpret_cast<CUdeviceptr>(buff);
+  bool cuMemAlloc = mscclpp::isCuMemMapAllocated((void*)dptr);
+  if (cuMemAlloc) {
+    int fd;
+    cuMemGetHandleForAddressRange(&fd, dptr, pages * pageSize, CU_MEM_RANGE_HANDLE_TYPE_DMA_BUF_FD, 0);
+    this->mr = IBVerbs::ibv_reg_dmabuf_mr2(pd, 0, pages * pageSize, addr, fd,
```

Comment:
Please double check the size here. I think we need to use another size instead of `pages * pageSize`